app/desktop/src/main/kotlin/AniDesktop.kt

```diff
@@ -437,14 +418,24 @@ object AniDesktop {
     }
 }
 
+@OptIn(InternalComposeUiApi::class)
 @Composable
 private fun FrameWindowScope.MainWindowContent(
     aniNavigator: AniNavigator,
 ) {
     val settingsRepository = KoinPlatform.getKoin().get<SettingsRepository>()
     AniApp {
         window.setTitleBar(AniThemeDefaults.navigationContainerColor, isSystemInDarkTheme())
-
+        val themeSettings = LocalThemeSettings.current
+        val titleBarThemeController = LocalTitleBarThemeController.current
+        val systemTheme = LocalSystemTheme.current
+        LaunchedEffect(themeSettings.darkMode, titleBarThemeController, systemTheme) {
+            titleBarThemeController?.isDark = when (themeSettings.darkMode) {
+                DarkMode.AUTO -> systemTheme == SystemTheme.Dark
+                DarkMode.LIGHT -> false
+                DarkMode.DARK -> true
+            }
+        }
```

Comment:
no need to use LaunchedEffect


```suggestion
        titleBarThemeController?.isDark = when (themeSettings.darkMode) {
            DarkMode.AUTO -> systemTheme == SystemTheme.Dark
            DarkMode.LIGHT -> false
            DarkMode.DARK -> true
        }
```